File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cs
public class Solution {
    public int[] TopKFrequent(int[] nums, int k) {
        
        Dictionary<int, int> count = new();
        foreach(int num in nums)
        {
            count[num] = count.GetValueOrDefault(num) + 1;
        }
        var sortedCount = count.OrderByDescending(n => n.Value).ToList();

        int[] ans = new int[k];
        for(int i = 0; i < k; i++) ans[i] = sortedCount[i].Key;
        return ans;
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        mp[0] = 1;
        int pref = 0;
        int ans = 0;
        for(int i = 0; i < nums.size(); i++)
        {
            pref += nums[i];

            if(mp.count(-k+pref))
            {
                ans += mp[-k+pref];
            }
            mp[pref]++;
        }   
        return ans;
    }
};